kubectl-fdb/cmd/root.go

```diff
@@ -175,3 +176,58 @@ func usingLatestPluginVersion(cmd *cobra.Command, pluginVersionChecker VersionCh
 
 	return nil
 }
+
+type processSelectionOptions struct {
+	ids               []string
+	namespace         string
+	clusterName       string
+	clusterLabel      string
+	processClass      string
+	useProcessGroupID bool
+}
+
+// getProcessGroupsByCluster returns a map of processGroupIDs by FDB cluster that match the criteria in the provided
+// processSelectionOptions.
+func getProcessGroupsByCluster(kubeClient client.Client, opts processSelectionOptions) (map[*fdbv1beta2.FoundationDBCluster][]fdbv1beta2.ProcessGroupID, error) {
```

Comment:
Most of those selection and "util" methods are in the `k8s_client` file: https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/kubectl-fdb/cmd/k8s_client.go.

Should we add dedicated tests for this method?

Reply:
not a bad idea, I think it's decently tested via the remove process group one right now but a basic set for it sounds good